tests/test_import.py

```diff
@@ -16,6 +17,92 @@ def assert_equal_dictionary(d1, d2):
             assert d1[k] == d2[k]
 
 
+def test_dedupe_parameters():
+    parameters = [
+        {'name': 'SigXsecOverSM', 'bounds': [[0.0, 10.0]]},
+        {'name': 'SigXsecOverSM', 'bounds': [[0.0, 10.0]]},
+    ]
+    assert len(pyhf.readxml.dedupe_parameters(parameters)) == 1
+    parameters[1]['bounds'] = [[0.0, 2.0]]
+    with pytest.raises(RuntimeError) as excinfo:
+        pyhf.readxml.dedupe_parameters(parameters)
+        assert 'SigXsecOverSM' in str(excinfo.value)
+
+
+def test_process_normfactor_configs():
+    # Check to see if mu_ttbar NormFactor is overridden correctly
+    # - ParamSetting has a config for it
+    # - other_parameter_configs has a config for it
+    # Make sure that when two measurements exist, we're copying things across correctly
+    toplvl = ET.Element("Combination")
+    meas = ET.Element(
+        "Measurement",
+        Name='NormalMeasurement',
+        Lumi=str(1.0),
+        LumiRelErr=str(0.017),
+        ExportOnly=str(True),
+    )
+    poiel = ET.Element('POI')
+    poiel.text = 'mu_SIG'
+    meas.append(poiel)
+
+    se = ET.Element('ParamSetting', Const='True')
+    se.text = ' '.join(['Lumi', 'mu_both', 'mu_paramSettingOnly'])
+    meas.append(se)
+
+    se = ET.Element('ParamSetting', Val='2.0')
+    se.text = ' '.join(['mu_both'])
+    meas.append(se)
+
+    toplvl.append(meas)
+
+    meas = ET.Element(
+        "Measurement",
+        Name='ParallelMeasurement',
+        Lumi=str(1.0),
+        LumiRelErr=str(0.017),
+        ExportOnly=str(True),
+    )
+    poiel = ET.Element('POI')
+    poiel.text = 'mu_BKG'
+    meas.append(poiel)
+
+    se = ET.Element('ParamSetting', Val='3.0')
+    se.text = ' '.join(['mu_both'])
+    meas.append(se)
+
+    toplvl.append(meas)
+
+    other_parameter_configs = [
+        dict(name='mu_both', inits=[1.0], bounds=[[1.0, 5.0]], fixed=False),
+        dict(name='mu_otherConfigOnly', inits=[1.0], bounds=[[0.0, 10.0]], fixed=False),
+    ]
+
+    result = pyhf.readxml.process_measurements(
+        toplvl, other_parameter_configs=other_parameter_configs
+    )
+    result = {
+        m['name']: {k['name']: k for k in m['config']['parameters']} for m in result
+    }
+    assert result
+    # make sure ParamSetting configs override NormFactor configs
+    assert result['NormalMeasurement']['mu_both']['fixed']
+    assert result['NormalMeasurement']['mu_both']['inits'] == [2.0]
+    assert result['NormalMeasurement']['mu_both']['bounds'] == [[1.0, 5.0]]
+    # make sure ParamSetting is doing the right thing
```

Comment:
To make these `assert` blocks visually easier to read can you separate them with a new line before the relevant comment?